champ/lib/storage/mock/mock.rs

```diff
@@ -75,8 +74,74 @@ impl Database for MockDB {
             let tx_id_str = hex::encode(&tx_id);
 
             account_transactions.push(tx_id_str.clone());
-            self.transactions.insert(tx_id_str, tx).ok_or(DatabaseError::Unknown)?;
+            self.transactions
+                .insert(tx_id_str, (tx, account_hash.clone()))
+                .ok_or(DatabaseError::Unknown)?;
         }
         Ok(())
     }
+
+    async fn get_block_by_height(&self, account_id: &str, block_height: &u64) -> Result<&api::Block, DatabaseError> {
+        self.blocks
+            .iter()
+            // reverse to make it faster for newer blocks
+            .rev()
+            .find_map(|b| {
+                if matches!(b.1.to_owned().data, Some(block) if block.address == account_id && &block.height == block_height) {
+                    Some(b.1)
+                } else {
+                    None
+                }
+            })
+            .ok_or(DatabaseError::Unknown)
+    }
+
+    async fn get_account_delegate(&self, _account_id: &str) -> Result<Option<String>, DatabaseError> {
+        let delegate = self
+            .transactions
+            .iter()
+            // rverse since only the newest transaction counts
+            .rev()
+            .find_map(|t| {
+                if let Some(TxDelegate(delegate_tx)) = &t.1 .0.data {
+                    return Some(Some(delegate_tx.representative.clone()));
+                }
+                None
+            })
+            .ok_or(DatabaseError::Unknown)?
+            .ok_or(DatabaseError::Unknown)?;
+
+        let delegate_str = hex::encode(delegate);
+        if delegate_str.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(delegate_str))
+        }
+    }
+
+    async fn get_delegates_by_account(&self, account_id: &str) -> Result<Vec<String>, DatabaseError> {
+        let mut delegated_accounts = HashSet::new();
+
+        let account_hex = hex::decode(account_id).map_err(|_| DatabaseError::Unknown)?;
+        let delegates = self
+            .transactions
+            .iter()
+            .rev()
+            .filter_map(|t| {
+                if let Some(TxDelegate(delegate_tx)) = &t.1 .0.data {
+                    if delegate_tx.representative == account_hex {
+                        // only the latest transaction counts per account
+                        if delegated_accounts.contains(&t.1 .1) {
+                            return None;
+                        }
+                        delegated_accounts.insert(t.1 .1.clone());
```

Comment:
I think delegate_accounts already has the accIDs we want, and the return inside Filter_map is wrong